base/statistics.jl

```diff
@@ -550,7 +550,7 @@ function quantile!(q::AbstractArray, v::AbstractVector, p::AbstractArray;
     end
     isnan(v[end]) && throw(ArgumentError("quantiles are undefined in presence of NaNs"))
 
-    for i = 1:length(p)
+    for i in eachindex(p)
         @inbounds q[i] = _quantile(v,p[i])
```

Comment:
You should use two different indices for `p` and `q`, with `zip(eachindex(p), eachindex(q))`.